File: Array/2373. Largest Local Values in a Matrix/largestLocal.go
func largestLocal(grid [][]int) [][]int {
	n := len(grid) - 2
	result := make([][]int, n)
	for i := range result {
		result[i] = make([]int, n)
	}
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			num := 0
			for a := i; a < i+3; a++ {
				for b := j; b < j+3; b++ {
					if grid[a][b] > num {
						num = grid[a][b]
					}
				}
			}
			result[i][j] = num
		}
	}
	return result
}

File: Array/2373. Largest Local Values in a Matrix/largestLocal.rs
impl Solution {
    pub fn largest_local(grid: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let n = grid.len() - 2;
        let mut result = vec![vec![0; n]; n];
        
        for i in 0..n {
            for j in 0..n {
                let mut num = 0;
                for a in i..i+3 {
                    for b in j..j+3 {
                        if grid[a][b] > num {
                            num = grid[a][b];
                        }
                    }
                }
                result[i][j] = num;
            }
        }
        
        result
    }
}